### PR TITLE
feat: cache template file

### PR DESCRIPTION
**Pull request by jleach**
# Summary of Changes

Cache the bundle template file for faster access and reliability.

# Screenshots, videos, or gifs

n/a

# Breaking change guide

n/a

# Related Issues

n/a

# Pull Request Checklist

Tick all boxes below to demonstrate that you have completed the respective task. If the item does not apply to your this PR **check it anyway** to make it apparent that there's nothing to do.

- [x] All commits contain a DCO `Signed-off-by` line (we use the [DCO GitHub app](https://github.com/apps/dco) to enforce this)
- [x] If applicable, screenshots, gifs, or video are included for UI changes
- [x] If applicable, breaking changes are described above along with how to address them
- [x] Updated documentation as needed for changed code and new or modified features
- [x] Added sufficient [tests](../packages/legacy/core/__tests__/) so that overall code coverage is not reduced

If you have _any_ questions to _any_ of the points above, just **submit and ask**! This checklist is here to _help_ you, not to deter you from contributing!

Pro Tip 🤓

- Read our [contribution guide](../CONTRIBUTING.md) at least once; it will save you a few review cycles!
- Your PR will likely not be reviewed until all the above boxes are checked and all automated checks have passed

### CODE REVIEW

File: packages/legacy/core/App/utils/proofBundle.ts [hunk 1]
```diff
@@ -87,76 +94,126 @@ export const useRemoteProofBundleResolver = (
...
     }
 
-    let templateData = this.templateData
+    templateData = this.templateData
 
     if (acceptDevRestrictions) {
       templateData = applyDevRestrictions(templateData)
     }
+
     const template = templateData.find((template) => template.id === templateId)
```
Comment (al-rosenthal) [inline packages/legacy/core/App/utils/proofBundle.ts]: nit: you could just return this value

**Review by al-rosenthal — COMMENTED** [review]
just a small nit pick and the latest changes need to be merged, otherwise looks good

File: packages/legacy/core/App/utils/proofBundle.ts (hunk 1 above)
Comment (jleach) [reply packages/legacy/core/App/utils/proofBundle.ts]: Fixed.

**Review by al-rosenthal — APPROVED** [review]
needs to merge latest but otherwise looks good!

**Review by al-rosenthal — APPROVED** [review]

**Review by bryce-mcmath — APPROVED** [review]

**Pull request merged**